### PR TITLE
[#IOPID-1514] Added missing SAMLResponse alert

### PR DESCRIPTION
**Pull request by arcogabbo**
### List of changes
* [[#IOPID-1514] Added missing SAMLResponse alert](https://github.com/pagopa/io-infra/commit/39e3e4bc9e1654e021dbcda218d2a22c845545a9)

### Motivation and context
Adding the alert for the new error message introduced in https://github.com/pagopa/io-spid-commons/pull/156

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### If PR is partially applied, why? (reserved to mantainers)
unwanted changes spotted on plan results, I am going to perform a local apply only for the alert resource to be created

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/domains/citizen-auth-app/01_monitor.tf [hunk 1]
```diff
@@ -31,3 +40,47 @@ data "azurerm_monitor_action_group" "email" {
...
+  name                    = "[${upper(var.domain)}] Missing required SAMLResponse in assertionConsumerService"
+  resource_group_name     = data.azurerm_resource_group.monitor_rg.name
+  scopes                  = [data.azurerm_application_gateway.app_gateway.id]
+  description             = <<-EOT
+    Detected multiple SAMLResponse missing during assertionConsumerService.
+    IdP is unknown here but you can watch Mixpanel events to identify the IdP
+    that is causing the alert to trigger
+  EOT
+  severity                = 0
+  auto_mitigation_enabled = false
```
Comment (gquadrati) [inline src/domains/citizen-auth-app/01_monitor.tf:55]: To allow the alert to be triggered once per event, we need to set `auto_mitigation_enabled` to `true`.

File: src/domains/citizen-auth-app/01_monitor.tf [hunk 2]
```diff
@@ -31,3 +40,47 @@ data "azurerm_monitor_action_group" "email" {
...
+    query                   = <<-QUERY
+AzureDiagnostics
+| where originalHost_s in ("app-backend.io.italia.it", "api-app.io.pagopa.it")
+| where requestUri_s == "/error.html"
+| where httpMethod_s == "GET"
+| where requestQuery_s contains "Missing%20SAMLResponse%20in%20ACS"
+    QUERY
+    operator                = "GreaterThanOrEqual"
+    time_aggregation_method = "Count"
+    threshold               = 5
```
Comment (gquadrati) [inline src/domains/citizen-auth-app/01_monitor.tf:73]: How did we decided the threshold?

File: src/domains/citizen-auth-app/01_monitor.tf [hunk 3]
```diff
@@ -31,3 +40,47 @@ data "azurerm_monitor_action_group" "email" {
...
+  enabled                 = true
+  name                    = "[${upper(var.domain)}] Missing required SAMLResponse in assertionConsumerService"
+  resource_group_name     = data.azurerm_resource_group.monitor_rg.name
+  scopes                  = [data.azurerm_application_gateway.app_gateway.id]
+  description             = <<-EOT
+    Detected multiple SAMLResponse missing during assertionConsumerService.
+    IdP is unknown here but you can watch Mixpanel events to identify the IdP
+    that is causing the alert to trigger
+  EOT
+  severity                = 0
```
Comment (gquadrati) [inline src/domains/citizen-auth-app/01_monitor.tf:54]: I'd go with severity 1.

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/domains/citizen-auth-app/01_monitor.tf (hunk 2 above)
Comment (arcogabbo) [reply src/domains/citizen-auth-app/01_monitor.tf:73]: some providers have very low login rate so I decided that 5 missing SAMLResponses could have been good for triggering the alert. do you believe I should lower/increment the threshold?

File: src/domains/citizen-auth-app/01_monitor.tf (hunk 1 above)
Comment (arcogabbo) [reply src/domains/citizen-auth-app/01_monitor.tf]: done in https://github.com/pagopa/io-infra/pull/942/commits/cb9617c2b3285ef5253dc9689216fbdd9a55e71e

**Review by gquadrati — APPROVED** [review]
LGTM

**Pull request merged**